node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -176,9 +180,11 @@ trait Node extends P2PLogger {
     } yield {
       peerMsgSenderF.map(_.sendGetHeadersMessage(hash.flip))
       logger.info(s"Starting sync node, height=${header.height} hash=$hash")
-    }
+    }).flatMap(_ => onSync())
```

Comment:
I don't think this actually guarantees this callback is called when we are synced. So this is a little misleading IMO. Basically this should be called `onStartSync()`.